File: csharp/AwqatSalah/Awqat/Prayer.cs
namespace Awqat;

/// <summary>PrayerTime (Namaz Vakitleri) endpoint'leri. İstemciyi ilk argüman alır.</summary>
public static class Prayer
{
    public static Task<List<PrayerTime>> DailyAsync(Client c, long cityId) =>
        c.GetJsonAsync<List<PrayerTime>>($"/api/PrayerTime/Daily/{cityId}");

    public static Task<List<PrayerTime>> WeeklyAsync(Client c, long cityId) =>
        c.GetJsonAsync<List<PrayerTime>>($"/api/PrayerTime/Weekly/{cityId}");

    public static Task<List<PrayerTime>> MonthlyAsync(Client c, long cityId) =>
        c.GetJsonAsync<List<PrayerTime>>($"/api/PrayerTime/Monthly/{cityId}");

    public static Task<List<PrayerTime>> RamadanAsync(Client c, long cityId) =>
        c.GetJsonAsync<List<PrayerTime>>($"/api/PrayerTime/Ramadan/{cityId}");

    public static Task<EidPrayerTime> EidAsync(Client c, long cityId) =>
        c.GetJsonAsync<EidPrayerTime>($"/api/PrayerTime/Eid/{cityId}");
}

File: js/awqat/prayer.js
// PrayerTime (Namaz Vakitleri) endpoint'leri. İstemciyi ilk argüman olarak alır.

export const daily = (c, cityId) => c.getJSON(`/api/PrayerTime/Daily/${cityId}`);
export const weekly = (c, cityId) => c.getJSON(`/api/PrayerTime/Weekly/${cityId}`);
export const monthly = (c, cityId) => c.getJSON(`/api/PrayerTime/Monthly/${cityId}`);
export const ramadan = (c, cityId) => c.getJSON(`/api/PrayerTime/Ramadan/${cityId}`);
export const eid = (c, cityId) => c.getJSON(`/api/PrayerTime/Eid/${cityId}`);
